### ISSUE TITLE
Details of training and implementation in VPL

### ISSUE THREAD

**Issue by hang5312** [deepinsight/insightface#1629]
Hi, I've read your recent work VPL and tried to use it in our project, I implemented Arcface with VPL in Mxnet and I have met several questions:
1. In the memory injection part,  the Dimensional of embedding in Memory bank (M) is significantly different with the weight (W, which is [0,1] ). In order to do the injection, I normalized M before adding it to W. And the final weight = **Norm( 0.15 * Norm(M)  + 0.85 * W )**  Is this assumption matches your implementation of VPL ?
2. Is there any ideas of hyper-parameter settings in different data distribution ? Which hyper-parameter is more important, t or M_ratio (which is decided by t)? Because in my datasets, I have to set t =2000 to reach M_ratio = 40% and memory bank might contain features from early steps.

Thanks !

**Comment by nttstar** [comment]
Hi, here is our torch based implementation, you can check it: https://github.com/deepinsight/insightface/tree/master/recognition/vpl

More docs is coming
